mula/scheduler/schedulers/schedulers/report.py

```diff
@@ -1,210 +1,125 @@
+    @exception_handler
+    @tracer.start_as_current_span("push_report_task")
+    def push_report_task(
+        self, report_task: models.ReportTask, organisation_id: str, create_schedule: bool, caller: str = ""
+    ) -> None:
         if self.is_item_on_queue_by_hash(report_task.hash):
-            self.logger.debug(
-                "Report task already on queue",
-                task_hash=report_task.hash,
-                organisation_id=self.organisation.id,
-                scheduler_id=self.scheduler_id,
-                caller=caller,
-            )
+            self.logger.debug("Report task already on queue", scheduler_id=self.scheduler_id, caller=caller)
             return
 
-        task = Task(
+        task = models.Task(
             scheduler_id=self.scheduler_id,
+            organisation=organisation_id,
             priority=int(datetime.now().timestamp()),
             type=self.ITEM_TYPE.type,
             hash=report_task.hash,
             data=report_task.model_dump(),
         )
 
-        try:
-            self.push_item_to_queue_with_timeout(task, self.max_tries)
-        except QueueFullError:
-            self.logger.warning(
-                "Could not add task %s to queue, queue was full",
-                report_task.hash,
-                task_hash=report_task.hash,
-                queue_qsize=self.queue.qsize(),
-                queue_maxsize=self.queue.maxsize,
-                organisation_id=self.organisation.id,
-                scheduler_id=self.scheduler_id,
-                caller=caller,
-            )
-            return
+        self.push_item_to_queue_with_timeout(task, self.max_tries)
 
         self.logger.info(
             "Report task pushed to queue",
             task_id=task.id,
-            task_hash=report_task.hash,
-            organisation_id=self.organisation.id,
+            task_hash=task.hash,
             scheduler_id=self.scheduler_id,
             caller=caller,
         )
 
-    def has_report_task_started_running(self, task: ReportTask) -> bool:
-        task_db = None
-        try:
-            task_db = self.ctx.datastores.task_store.get_latest_task_by_hash(task.hash)
-        except storage.errors.StorageError as exc_db:
-            self.logger.error(
-                "Could not get latest task by hash %s",
-                task.hash,
-                organisation_id=self.organisation.id,
-                scheduler_id=self.scheduler_id,
-                exc_info=exc_db,
-            )
-            raise exc_db
-
-        if task_db is not None and task_db.status not in [TaskStatus.FAILED, TaskStatus.COMPLETED]:
-            self.logger.debug(
-                "Task is still running, according to the datastore",
-                task_id=task_db.id,
-                organisation_id=self.organisation.id,
-                scheduler_id=self.scheduler_id,
-            )
-            return True
+    def has_schedule_permission_to_run(self, schedule: models.Schedule) -> bool:
```

Comment:
Can we somehow move this to the original query that pops our schedules, instead of evaluating this in python time?

Reply:
original method has been removed